### PR TITLE
B2B renamed to CORPORATE as this is the actual barrier type name

### PR DESCRIPTION
**Pull request by ifyio**
cc/ @commuterjoy @aintgoin2goa

### CODE REVIEW

File: src/barrierAPIClient.js [hunk 1]
```diff
@@ -77,12 +95,18 @@ function getBarrierData(barrierType, requestHeaders, flags){
...
-
+	function fetchCorporateDataIfApplicable() {
+		if(barrierType === 'CORPORATE') {
+			return fetch('http://barrier-app.memb.ft.com/memb/amo/v1/amo-data', {
+				headers: {
+					'True-Client-Ip': requestHeaders['True-Client-Ip']
+				}
+			})
+			.then(response => {
+				return response.ok ? response.json() : undefined;
```
Comment (aintgoin2goa) [inline src/barrierAPIClient.js]: You're returning a promise OR undefined?  Is this gonna cause an issue later on?  If so might be better to replace `undefined` with `Promise.resolve(undefined)`?

Comment (matthew-andrews) [inline src/barrierAPIClient.js]: is it not safe to return `undefined`?

File: src/barrierAPIClient.js [hunk 2]
```diff
@@ -70,6 +70,24 @@ function getBarrierData(barrierType, requestHeaders, flags){
 	var url = endpoints.prodDirect;
 	var cacheKey = {url:url, headers:requestHeaders};
 
+	return Promise.resolve()
```
Comment (matthew-andrews) [inline src/barrierAPIClient.js]: it doesn't really matter but I think you could start with your `Promise.all` and skip this `Promise.resolve()`?

```js
return Promise.all([
    fetchBarrierData(),
    fetchCorporateDataIfApplicable()
  ])
  .then(results => {
     // etc
``

**Comment by aintgoin2goa** [comment]
:+1:

File: src/barrierAPIClient.js (hunk 2 above)
Comment (ifyio) [inline src/barrierAPIClient.js]: Yes good spot Matt. Not sure what I was thinking when I wrote that line but that's what I would have done as well (if i was in my right mind)

**Pull request merged**